verboseTree-maven-plugin/src/main/java/org/apache/maven/dependency/graph/DependencyGraphBuilder.java

```diff
@@ -39,12 +57,118 @@
+        File file = new File( project.getBasedir().getAbsolutePath().replace( '\\', '/' ) + "/target/tree.txt" );
+        /*SerializeGraph serializer = new SerializeGraph();
+        String serialized = serializer.serialize( rootNode );*/
+        try
+        {
+            write( file, rootNode.toString() + rootNode.getChildren().get( 0 ).getDependency().getScope() );
+            /*write( file, "why" + rootNode.toString() + " " + rootNode.getChildren().get( 0 ).toString()
+            + " " + rootNode.getChildren().get( 1 ).toString() + "\n" +
+                    rootNode.getChildren().get( 2 ).toString() + " " + rootNode.getChildren().get( 3 ).toString()
+            + " " + rootNode.getChildren().get( 0 ).getChildren().size()
+                    + " " + rootNode.getChildren().get( 1 ).getChildren().size()
+                    + " " + rootNode.getChildren().get( 2 ).getChildren().size()
+                    + " " + rootNode.getChildren().get( 3 ).getChildren().size());*/
+            // write(file, serialized);
+        }
+        catch ( IOException e )
+        {
+            e.printStackTrace();
+            getLog().error( "Failed to write to file:" + file.getAbsolutePath() );
+        }
+    }
+
+    public DependencyNode buildDependencyGraph() throws Exception
+    {
+        // adapting the dependency-plugin code
+        ProjectBuildingRequest buildingRequest =
+                new DefaultProjectBuildingRequest( session.getProjectBuildingRequest() );
+
+        buildingRequest.setProject( project );
+
+        // need to configure the repositorySystemSession
+
+        // dependency plugin code that isn't needed below
+        // dependencyGraphBuilder.buildDependencyGraph( buildingRequest, artifactFilter, reactorProjects );
+
+        // now adapting the dependency-tree defaultDependencyGraphBuilder and maven31Code
+
+        final DependencyResolutionRequest request = new DefaultDependencyResolutionRequest();
+        request.setMavenProject( project );
+        request.setRepositorySession( session.getRepositorySession() );
+        // request.setRepositorySession( repositorySystemSession );
+
+        final DependencyResolutionResult result = resolveDependencies( request, null );
+        DependencyNode graphRoot = result.getDependencyGraph();
+
+
+        return graphRoot;
+    }
+
+
+    public DependencyNode getDependencyGraph()
+    {
+        return rootNode;
+    }
+
+
+    private DependencyResolutionResult resolveDependencies( DependencyResolutionRequest request,
+                                                            Collection<MavenProject> reactorProjects )
+            throws Exception
+    {
+        try
+        {
+            return resolver.resolve( request );
+        }
+        catch ( DependencyResolutionException e )
+        {
+            if ( reactorProjects == null )
+            {
+                throw new Exception( "Could not resolve following dependencies: "
```

Comment:
change this to a more specific exception

Reply:
Should I copy the DependencyGraphBuilderException in the maven-dependency-tree plugin and use that?

Reply:
DependencyResolutionException is probably what you need here